src/server/services/procedures/paleocean-oxygen-isotopes/paleocean-oxygen-isotopes.js

```diff
@@ -0,0 +1,120 @@
+};
+
+const importDelta18OData = function(line) {
+    let [time, d18O, error] = line.trim().split(/\s+/);
+    [time, d18O, error].forEach(value=>assert(!isNaN(Number(value))));
+    let adjustTime = timeConversion(time);
+    PaleoceanOxygenIsotopes._dataDelta18O.push({adjustTime, d18O, error});
+};
+
+const importSedimentationRatesData = function(line) {
+    let [time, averagedSedRates, normalizedSedRates] = line.trim().split(/\s+/);
+    [time, averagedSedRates, normalizedSedRates].forEach(value=>assert(!isNaN(Number(value))));
+    let adjustTime = timeConversion(time);
+    PaleoceanOxygenIsotopes._dataSedim.push({adjustTime, averagedSedRates, normalizedSedRates});
+};
+
+fs.readFileSync(path.join(__dirname, 'Pliocene-Pleistocene Benthic d18O Stack.txt'), 'utf8')
+    .split('\n')
+    .forEach(line=>importDelta18OData(line));
+
+fs.readFileSync(path.join(__dirname, 'Sedimentation Rates.txt'), 'utf8')
+    .split('\n')
+    .forEach(line=>importSedimentationRatesData(line));
+
+/**
+ * Get delta 18O value (unit: per mill. It is a parts per thousand unit, often used directly to 
+ * refer to isotopic ratios and calculated by calculating the ratio of isotopic concentrations in 
+ * a sample and in a standard, subtracting one and multiplying by one thousand).
+ *
+ * If a start and end year is provided, only measurements within the given range will be
+ * returned.
+ *
+ * @param {Number=} startyear Year to begin data at
+ * @param {Number=} endyear Year to begin data at
+ * @returns {Array} delta 18O
+ */
+PaleoceanOxygenIsotopes.getDelta18O = function(startyear = -Infinity, endyear = Infinity) {
+    return this._dataDelta18O
+        .map(data => [data.adjustTime, data.d18O])
+        .filter(data => data[0] >= startyear && data[0] <= endyear);
+};
+
+/**
+ * Get delta 18O error value (unit: per mill).
+ *
+ * If a start and end year is provided, only measurements within the given range will be
+ * returned.
+ *
+ * @param {Number=} startyear Year to begin data at
+ * @param {Number=} endyear Year to begin data at
+ * @returns {Array} delta 18O error
+ */
+PaleoceanOxygenIsotopes.getDelta18OError = function(startyear = -Infinity, endyear = Infinity) {
+    return this._dataDelta18O
+        .map(data => [data.adjustTime, data.error])
+        .filter(data => data[0] >= startyear && data[0] <= endyear);
+};
+
+/**
+ * Get average sedimentation rate value (unit: centimeter per kiloyear).
+ *
+ * If a start and end year is provided, only measurements within the given range will be
+ * returned.
+ *
+ * @param {Number=} startyear Year to begin data at
+ * @param {Number=} endyear Year to begin data at
+ * @returns {Array} average sedimentation rate
+ */
+PaleoceanOxygenIsotopes.getAverageSedimentationRates = function(startyear = -Infinity, endyear = Infinity) {
+    return this._dataSedim
+        .map(data => [data.adjustTime, data.averagedSedRates])
+        .filter(data => data[0] >= startyear && data[0] <= endyear);
```

Comment:
This actually would be more legible (and slightly more performant) if you first filter than map:
```javascript
return this._dataSedim
    .filter(data => data.adjustTime >= startyear && data.adjustTime <= endyear)
    .map(data => [data.adjustTime, data.averagedSedRates]);
```

Reply:
Also, maybe just recording the time as `year` (or even `yearCE`) rather than `adjustTime` might be a little more clear since it doesn't really matter that it had to be adjusted for the downstream usage (like here).